src/agent/src/message_queue_utils.cpp

```diff
@@ -48,3 +50,44 @@ void PushCommandsToQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue, const
         }
     }
 }
+
+std::optional<command_store::Command> GetCommandFromQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue)
+{
+    if (multiTypeQueue->isEmpty(MessageType::COMMAND))
+    {
+        return std::nullopt;
+    }
+
+    Message m = multiTypeQueue->getNext(MessageType::COMMAND);
+    nlohmann::json jasonData = m.data.at(0).at("data");
+
+    std::string strId = jasonData["id"].dump();
+
+    // remove quotation marks from id
+    strId.erase(strId.begin());
+    strId.erase(strId.end() - 1);
+    int64_t id = std::stoll(strId);
```

Comment:
I think the ID could be a string. We shouldn't assume it will be only numbers.

Reply:
Done